pid_tracker.py

```diff
@@ -161,18 +186,40 @@ def update_state(self, state):
         try:
           process = psutil.Process(int(pid))
           state.set_up((time.time() - process.create_time) * 1000)
+
+          meminfo = process.get_ext_memory_info()
+          state.rss = meminfo.rss
+          state.shared_mem = meminfo.shared
         except Exception, e:
           state.set_down()
           self.collectd.debug('pid-tracker plugin: pid for pidfile does not point at running process. PidFile=%s, pid=%s. Exception=%s' % (state.pid_file, pid, e))
 
-  def create_metric(self, pid_state, extra_dimensions=""):
+  def dispatch_metrics(self, pid_state, extra_dimensions=""):
     self.log_verbose('Sending value counter.%s[plugin_instance=%s]=%s, extra_dimensions: %s' % (UPTIME_METRIC, pid_state.plugin_instance, pid_state.uptime, extra_dimensions))
-    return self.collectd.Values(
+    self.collectd.Values(
       plugin=PLUGIN,
       plugin_instance=pid_state.plugin_instance,
       type="counter", 
       type_instance="%s%s" % (UPTIME_METRIC, extra_dimensions),
-      values=[pid_state.uptime])
+      values=[pid_state.uptime]).dispatch()
+
+    print "dispatching metrics %s\n" % (pid_state.collect_mem_stats)
+    if pid_state.collect_mem_stats:
```

Comment:
Can we also not ship these metrics if the process is down? Will cut down on some DPM maybe